feat: Implemented Data Spotlight Section Pattern

muhammad-ali-pk opened the pull request:
## Done

- Implemented Data Spotlight Pattern
- Drive by: fixed newsletter pattern's default addendum slot not rendering
- Drive by: added a new `submit_btn_class` param on the newsletter pattern

Fixes [WD-22552](https://warthogs.atlassian.net/browse/WD-22552) [WD-26620](https://warthogs.atlassian.net/browse/WD-26620)

## QA

- Review [Data spotlight pattern docs](https://vanilla-framework-5639.demos.haus/docs/patterns/data-spotlight)
- Review [release notes](https://vanilla-framework-5639.demos.haus/docs/whats-new)
- Review [combined examples](https://vanilla-framework-5639.demos.haus/docs/examples/patterns/data-spotlight/combined) and verify the pattern is structured as per the [Figma](https://www.figma.com/design/1brHCsyeTE6FU7hfJd6ao2/%F0%9F%9A%A7---WIP--Sites---Core-component-library?node-id=2471-36506)
- Review [newsletter signup pattern's combined examples](https://vanilla-framework-5639.demos.haus/docs/examples/patterns/newsletter-signup/combined) and verify the pattern is rendering default `description`, `addendum`, `hidden_fields` and `honeypot_fields` slots.
- Review [newsletter signup pattern's documentation page](https://vanilla-framework-5639.demos.haus/docs/patterns/newsletter-signup#jinja-macro) for newly added `submit_btn_class` param.

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) or macros should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).


[WD-22552]: https://warthogs.atlassian.net/browse/WD-22552?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WD-26620]: https://warthogs.atlassian.net/browse/WD-26620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

samhotep commented on releases.yml:
```diff
@@ -1,9 +1,23 @@
+      url: /docs/patterns/data-spotlight
+      status: New
+      notes: We've introduced a new Data spotolight pattern
 - version: 4.32.0
```
@muhammad-ali-pk Does this entry affect previous releases since `v4.33.0` has already been released?

samhotep commented on templates/docs/patterns/data-spotlight/index.md:
````diff
@@ -0,0 +1,213 @@
+
+```json
+{
+  "stat": "text (required)",
````
Is the `required` attributed supposed to be inside or outside the string? I see on line 46 it is not
```suggestion
  "stat": "text", (required)
```

samhotep commented on templates/docs/patterns/data-spotlight/index.md:
````diff
@@ -0,0 +1,213 @@
+```json
+{
+  "stat": "text (required)",
+  "headline": "text (optional)",
````
Here as well
```suggestion
  "headline": "text", (optional)
```

samhotep commented:
@muhammad-ali-pk the data spotlight looks fantastic! I've just left a few minor comments

- [ ] Where to put the type indicatiors. [e.g](https://github.com/canonical/vanilla-framework/pull/5639#discussion_r2348970892) 
- [ ] The padding when there's no description seems a bit different than figma, see
<img width="400" height="260" alt="image" src="https://github.com/user-attachments/assets/045afcf4-1584-4ebd-9212-3cc598b20e25" />
vs
<img width="400" height="260" alt="image" src="https://github.com/user-attachments/assets/bf6396e6-a815-42b2-8126-7283aeee70eb" />

muhammad-ali-pk commented:
@samhotep Thank you for the review! I have addressed your comments.

> The padding when there's no description seems a bit different than figma, see

@kim-isaac Could you please cross-check this? I believe the height of link on figma is different than the default height of link in vanilla (maybe because of the link-group block you used in figma?)

kim-isaac commented:
Hi @muhammad-ali-pk ! True, I adjusted the Figma component to match Vanilla. It’s now aligned with the demo. Thanks for letting me know!
<img width="880" height="562" alt="image" src="https://github.com/user-attachments/assets/66e1e92a-ca1c-40fd-9c90-497611a8ff02" />
